akka-http2-support/src/main/scala/akka/http/impl/engine/http2/FrameEvent.scala

```diff
@@ -71,3 +73,10 @@ final case class ParsedHeadersFrame(
   keyValuePairs: Seq[(String, String)],
   priorityInfo:  Option[PriorityFrame]
 ) extends FrameEvent
+
+// ------ internal frames, used for communicating settings between stages of our pipeline ----- 
+/**
+ * Special synthetic FrameEvent used to communicate enoder setting changes back to encoder.
+ * MUST NOT be rendered to network.
+ */
+private[http] final case class SyntheticHpackEncoderSettingFrame(setting: Setting) extends SyntheticFrameEvent
```

Comment:
Either we do this dance with sending back "down" the setting in a special one like that, or we need to make the HPack stuff a bidistage the same way like the framing was done now.